From 18298ffaf4c56fc170fe25fa03f35dc805b1916b Mon Sep 17 00:00:00 2001
From: Hariharan K <quic_harihk@quicinc.com>
Date: Wed, 16 Aug 2023 20:25:39 +0530
Subject: [PATCH] arch: qcom: Add a new kconfig to use different DDR profiles

This change adds a new config IPQ_MEM_PROFILE which can be used
to select different DDR profiles for different DDR size,
i.e 256MB, 512MB etc.
It also adds SKB_FIXED_SIZE_2K config which can be used to fix
'skbuf' size to 2KB.

Change-Id: I5707fcd23da3ef9734a2022d616bea922c2045ea
Signed-off-by: Hariharan K <quic_harihk@quicinc.com>
---
 arch/arm/mach-qcom/Kconfig | 17 +++++++++++++++++
 scripts/Makefile.lib       |  9 +++++++++
 2 files changed, 26 insertions(+)

diff --git a/arch/arm/mach-qcom/Kconfig b/arch/arm/mach-qcom/Kconfig
index 4fe85667d602..6a8be97e905a 100644
--- a/arch/arm/mach-qcom/Kconfig
+++ b/arch/arm/mach-qcom/Kconfig
@@ -47,4 +47,21 @@ config ARCH_MDM9615
 	bool "Enable support for MDM9615"
 	select CLKSRC_QCOM
 
+config IPQ_MEM_PROFILE
+	int "Select Memory Profile"
+	range 0 1024
+	default 0
+	help
+	  This option select memory profile to be used, which defines
+	  the reserved memory configuration used in device tree.
+
+	  If unsure, say 0
+
+config SKB_FIXED_SIZE_2K
+	bool "SKB size fixed at 2K"
+	default n
+	help
+	  This is a hint to the NSS driver that the ‘skbuff’ size might
+	  need to be fixed at 2KB, to conserve memory.
+
 endif
diff --git a/scripts/Makefile.lib b/scripts/Makefile.lib
index dc83d1f0deac..ff4cded170b4 100644
--- a/scripts/Makefile.lib
+++ b/scripts/Makefile.lib
@@ -248,6 +248,15 @@ dtc_cpp_flags  = -Wp,-MMD,$(depfile).pre.tmp -nostdinc                    \
 		 $(addprefix -I,$(DTC_INCLUDE))                          \
 		 -undef -D__DTS__
 
+#Add DDR profiling for different DDR size in dtsi
+ifeq ($(CONFIG_IPQ_MEM_PROFILE),256)
+dtc_cpp_flags += -D __IPQ_MEM_PROFILE_256_MB__
+else ifeq ($(CONFIG_IPQ_FLASH_16M_PROFILE),y)
+dtc_cpp_flags += -D __IPQ_MEM_PROFILE_256_MB__
+else ifeq ($(CONFIG_IPQ_MEM_PROFILE),512)
+dtc_cpp_flags += -D __IPQ_MEM_PROFILE_512_MB__
+endif
+
 ifeq ($(CONFIG_CPU_THERMAL),y)
 dtc_cpp_flags += -D __CPU_THERMAL__
 endif
-- 
2.34.1

